### PR TITLE
Create Tables in a Txn.

### PR DESCRIPTION
**Pull request by sramamoorthy73**
Make the check, and INSERT/UPDATE in a single txn. This is needed, when DB is opened in a shared cache mode.

### CODE REVIEW

File: listener.go [hunk 1]
```diff
@@ -61,29 +61,40 @@ func processSnapshot(snapshot *common.Snapshot) {
...
 		if err.Error() == "duplicate column name: last_sequence" {
+			tx.Rollback()
 			return
 		} else {
-			log.Error("[[" + err.Error() + "]]")
-			log.Panicf("Unable to create last_sequence column on DB.  Unrecoverable error ", err)
+			log.Panicf("Unable to create last_sequence column on DB.  Error {%v}", err.Error())
 		}
 	}
+
```
Comment (haomingzhang) [inline listener.go]: you can `defer prep.Close()` here.

File: listener.go [hunk 2]
```diff
@@ -61,29 +61,40 @@ func processSnapshot(snapshot *common.Snapshot) {
...
 	var numApidClusters int
-	apidClusters, err := db.Query("SELECT COUNT(*) FROM edgex_apid_cluster")
+	tx, err := db.Begin()
 	if err != nil {
-		log.Panicf("Unable to read database: %s", err.Error())
+		log.Panicf("Unable to open DB txn: {%v}", err.Error())
 	}
-	apidClusters.Next()
-	err = apidClusters.Scan(&numApidClusters)
+
```
Comment (haomingzhang) [inline listener.go]: you can `defer tx.Rollback()` here.

File: listener.go [hunk 3]
```diff
@@ -61,29 +61,40 @@ func processSnapshot(snapshot *common.Snapshot) {
...
+	_, err = prep.Exec()
+	if err != nil {
+		log.Errorf("Snapshot processing DB exec failed. Err: {%v}", err)
+		prep.Close()
+		tx.Rollback()
+		return
+	}
+	prep.Close()
+	tx.Commit()
```
Comment (haomingzhang) [inline listener.go]: We should check `err = tx.Commit()`

File: listener.go (hunk 2 above)
Comment (sramamoorthy73) [reply listener.go]: I prefer to be explicit for this case.

File: listener.go (hunk 3 above)
Comment (sramamoorthy73) [reply listener.go]: yes.

**Pull request merged**